src/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.cs

```diff
@@ -13,48 +13,21 @@
 ** 
 ===========================================================*/
 
-using System;
 using System.Globalization;
 using System.Runtime.CompilerServices;
 using System.Diagnostics;
 using System.Runtime.Versioning;
 
 namespace System.Collections.Generic
 {
-    #region ArraySortHelper for single arrays
-
     internal interface IArraySortHelper<TKey>
     {
         void Sort(TKey[] keys, int index, int length, IComparer<TKey> comparer);
         int BinarySearch(TKey[] keys, int index, int length, TKey value, IComparer<TKey> comparer);
     }
 
-    internal static class IntrospectiveSortUtilities
-    {
-        // This is the threshold where Introspective sort switches to Insertion sort.
-        // Empirically, 16 seems to speed up most cases without slowing down others, at least for integers.
-        // Large value types may benefit from a smaller number.
-        internal const int IntrosortSizeThreshold = 16;
-
-        internal static int FloorLog2PlusOne(int n)
-        {
-            int result = 0;
-            while (n >= 1)
-            {
-                result++;
-                n = n / 2;
-            }
-            return result;
-        }
-
-        internal static void ThrowOrIgnoreBadComparer(Object comparer)
-        {
-            throw new ArgumentException(SR.Format(SR.Arg_BogusIComparer, comparer));
-        }
-    }
-
     [TypeDependencyAttribute("System.Collections.Generic.GenericArraySortHelper`1")]
-    internal class ArraySortHelper<T>
+    internal partial class ArraySortHelper<T>
         : IArraySortHelper<T>
     {
         private static volatile IArraySortHelper<T> defaultArraySortHelper;
```

Comment:
Please rename `defaultArraySortHelper` to `s_defaultArraySortHelper` since it also has this `s_` name in corert